cy43439.go

```diff
@@ -240,8 +242,87 @@ func (d *Device) wifiLinkStatus() int32 {
 	return whd.CYW43_LINK_DOWN
 }
 
+func (d *Device) handleAsyncEvent(payload []byte) error {
+	as, err := whd.ParseAsyncEvent(payload)
+	if err != nil {
+		return err
+	}
+	return d.processAsyncEvent(as)
+}
+
+// ref: void cyw43_ll_process_packets(cyw43_ll_t *self_in)
+func (d *Device) processPackets() error {
```

Comment:
So if I understand correctly this code was extracted from `doIoctl`- did the extraction benefit us? I only see it used in `poll`. I'm very worried we just swapped out linear, singlethreaded code for spaghetti concurrent code when there was no real underlying problem to be solved by doing so.

Reply:
In the C driver, there are three places where the process_packets code is repeated:

cyw43_ll_process_packets()    // Rx recv and events, from cyw43_poll_func
cyw43_do_ioctl()              // Ioctl completions
cyw43_sdpcm_send_common()     // Tx send completions

All three instances do the same thing:

cyw43_ll_sdpcm_poll_device()
   cyw43_cb_process_async_event()
   etc...

So the new processPackets() is the replacement for those three code repeats and we handle all async processing in one place: poll().